### PR TITLE
add an openssl master container for easier testing

### PR DESCRIPTION
**Pull request by reaperhulk**
I don't want to add this into CI (yet), but it'd be nice to have a container that contains a daily OpenSSL master build that we can test with as we desire.

If we decide we want to merge this we should not merge until after I create the dockerhub repo

### CODE REVIEW

File: runners/stretch-openssl-master/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    git \
+    curl
+
+RUN curl -sSL https://bootstrap.pypa.io/get-pip.py | python
+
+RUN pip install -q tox
+
+WORKDIR /root
+
+RUN git clone --depth=1 https://github.com/openssl/openssl openssl
```
Comment (alex) [inline runners/stretch-openssl-master/Dockerfile]: You can drop teh final `openssl`

File: runners/stretch-openssl-master/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+RUN cd openssl && \
+    ./config --prefix=/opt/pyca/cryptography/openssl \
```
Comment (alex) [inline runners/stretch-openssl-master/Dockerfile]: Are there any config flags we should be passing?

Comment (reaperhulk) [inline runners/stretch-openssl-master/Dockerfile]: Most of the defaults are pretty good now, but we can add `no-comp` still (that's what we do on our Windows builds for the wheels)

Comment (alex) [inline runners/stretch-openssl-master/Dockerfile]: Is `no-comp` really not the default? (And why don't we do it on macOS)

Comment (reaperhulk) [inline runners/stretch-openssl-master/Dockerfile]: `no-comp` is not default, although compression is disabled entirely in TLS in 1.1.0 and above I believe?

On macOS we don't build our own. https://github.com/Homebrew/homebrew-core/blob/master/Formula/openssl@1.1.rb#L29

**Pull request merged**